Documentation/Testing/ProjectTesting.rst

```diff
@@ -6,262 +6,234 @@
+here. If you are using another structure, you have to adjust some scripts.
+
+*   :ref:`TYPO3 installation with Composer <t3start:install>` and default paths
+*   :ref:`DDEV <t3start:installation-ddev-tutorial>` is used for local development
+*   Local extensions and the site package are situated in :path:`packages`
+
+..  directory-tree::
+    :caption: Initial directory tree in version control
+    :level: 1
+
+    *   :path:`.ddev`
+    *   :path:`config`
+    *   :path:`packages`
+    *   :file:`composer.json`
+    *   :file:`composer.lock`
+
+The :file:`composer.json` looks like this:
+
+..  literalinclude:: _ProjectTesting/_composer.json
+    :caption: Example project composer.json before testing
+
+..  _testing-projects-installation:
+
+Install testing dependencies
+============================
+
+As a bare minimum it is suggested to use
+
+*   One coding style fixer for PHP, for example :composer:`friendsofphp/php-cs-fixer`
+*   One static code analyzer for PHP, for example :composer:`phpstan/phpstan`
+
+Depending on the complexity of your project you might need:
+
+*   :composer:`phpunit/phpunit`, if there is any PHP code of a complexity that
+    should be tested.
+*   Testing of scss, TypeScript or JavaScript (not covered here)
+*   Linting of YAML, XML, TypoScript (not covered here)
+*   :ref:`Writing acceptance tests <t3coreapi:testing-writing-acceptance>`
+
+You can install all these tools as development dependencies. They will then not
+be installed on your production system when Composer is executed with option
+`--no-dev` during deployment:
+
+..  code-block:: bash
+    :caption: Composer installation **during deployment**
+
+    composer install --no-dev
+
+For TYPO3 project you can use the package :composer:`typo3/coding-standards`
+which bundles :composer:`friendsofphp/php-cs-fixer`, :composer:`phpstan/phpstan`,
+:composer:`phpunit/phpunit` and some other helpful tools and comes with useful
+configurations for these.
+
+..  code-block:: bash
+    :caption: Require development dependencies
+
+    composer req --dev typo3/coding-standards
+
+If you want to do Unit or Functional tests on project level you also need the
+TYPO3 testing framework:
+
+..  code-block:: bash
+    :caption: Require development dependencies
+
+    composer req --dev typo3/coding-standards typo3/testing-framework
+
+.. _testing-projects-configuration:
+
+Test configuration on project level
+===================================
+
+We suggest to keep all project level test configuration in a common place that
+can be excluded from deployment. The Core uses a folder called :path:`Build` with
+one folder per test-type and we will follow that scheme here. If you put
+the configuration in other directories, adjust your configuration files
+accordingly.
+
+..  _testing-projects-configuration-cs:
+
+Code style tests and fixing
+---------------------------
+
+:composer:`typo3/testing-framework` comes with a predefined configuration for
```

Comment:
testing-framework cs-fixer configuration is not meant as template to be used. It's purly for that repository. We do not maintain it in regards for other projects or extensions.

Either move coding style to a dedicated section describing all possibilites side by side - or at least stay with one way in this document. Above, the typo3/coding-standards are required - but the config from testing framework should be used ? That does not make much sence. Or do I miss here something ?

Reply:
Sorry, mixed them up, fixed it.